### PR TITLE
refactor(interpreter): try remove InterceptorInterpreter

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

refactor(interpreter): try remove InterceptorInterpreter

Fixes #issue

### CODE REVIEW

File: src/query/storages/preludes/src/system/log_queue.rs [hunk 1]
```diff
@@ -0,0 +1,225 @@
+use std::any::Any;
```
Comment (BohuTANG) [inline src/query/storages/preludes/src/system/log_queue.rs:1]: License header :)

**Review by Xuanwo — APPROVED** [review]
Nice refactor!

**Comment by Xuanwo** [comment]
Run `taplo fmt` to make sure `Cargo.toml` formatted correctly.

**Review by youngsofun — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

File: src/query/service/src/interpreters/interpreter.rs [hunk 2]
```diff
@@ -45,26 +49,45 @@ pub trait Interpreter: Sync + Send {
 
     /// The core of the databend processor which will execute the logical plan and get the DataBlock
     async fn execute(&self, ctx: Arc<QueryContext>) -> Result<SendableDataBlockStream> {
-        let mut build_res = self.execute2().await?;
+        log_query_start(&ctx);
+
+        let mut build_res = match self.execute2().await {
```
Comment (sundy-li) [inline src/query/service/src/interpreters/interpreter.rs:54]: `execute2` could be renamed into `execute_impl`

Comment (zhang2014) [reply src/query/service/src/interpreters/interpreter.rs:54]: It will be rename to execute in the other pr.

**Pull request merged**